domain-management/src/main/java/org/jboss/as/domain/management/audit/AuditLogLoggerResourceDefinition.java

```diff
@@ -153,25 +154,11 @@ public void execute(final OperationContext context, final ModelNode operation) t
                         auditLoggerProvider.setLogReadOnly(AuditLogLoggerResourceDefinition.LOG_READ_ONLY.resolveModelAttribute(context, model).asBoolean());
                         boolean enabled = AuditLogLoggerResourceDefinition.ENABLED.resolveModelAttribute(context, model).asBoolean();
                         final AuditLogger.Status status = enabled ? AuditLogger.Status.LOGGING : AuditLogger.Status.DISABLED;
-                        // Change the logger status in a new step to give any subsequent handler adds a chance
-                        // to run before we flush any queued up log records
-                        context.addStep(new OperationStepHandler() {
-                            @Override
-                            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                        context.completeStep((OperationContext.ResultAction resultAction, OperationContext context1, ModelNode operation1) -> {
+                            if(resultAction == OperationContext.ResultAction.KEEP) {
                                 auditLoggerProvider.setLoggerStatus(status);
-                                context.completeStep(new OperationContext.RollbackHandler() {
-                                    @Override
-                                    public void handleRollback(OperationContext context, ModelNode operation) {
-                                        auditLoggerProvider.setLoggerStatus(oldStatus);
-                                    }
-                                });
-                            }
-                        }, OperationContext.Stage.RUNTIME);
-
-
-                        context.completeStep(new OperationContext.RollbackHandler() {
-                            @Override
-                            public void handleRollback(OperationContext context, ModelNode operation) {
+                            } else {
+                                auditLoggerProvider.setLoggerStatus(oldStatus);
```

Comment:
You shouldn't need this since if this else block is entered this OSH is not calling setLoggerStatus anywhere else.